workloads/cyclictest/common.sh

```diff
@@ -0,0 +1,272 @@
+    done
+  fi
+}
+
+deploy_operator() {
+  deploy_benchmark_operator ${OPERATOR_REPO} ${OPERATOR_BRANCH}
+  if [[ $? != 0 ]]; then
+     exit 1
+  fi
+  rm -rf benchmark-operator
+}
+
+run_workload() {
+  log "Deploying benchmark"
+  local TMPCR=$(mktemp)
+  envsubst < $1 > ${TMPCR}
+  run_benchmark ${TMPCR} ${TEST_TIMEOUT}
+  local rc=$?
+  if [[ ${TEST_CLEANUP} == "true" ]]; then
+    log "Cleaning up benchmark"
+    kubectl delete -f ${TMPCR}
+  fi
+  return ${rc}
+}
+
+check_logs_for_errors() {
+client_pod=$(oc get pods -n benchmark-operator --no-headers | awk '{print $1}' | grep cyclictest | awk 'NR==1{print $1}')
+if [ ! -z "$client_pod" ]; then
+  num_critical=$(oc logs ${client_pod} -n benchmark-operator | grep CRITICAL | wc -l)
+  if [ $num_critical -gt 3 ] ; then
+    log "Encountered CRITICAL condition more than 3 times in cyclictest logs"
+    log "Log dump of cyclictest pod"
+    oc logs $client_pod -n benchmark-operator
+    delete_benchmark
+    exit 1
+  fi
+fi
+}
+
+wait_for_benchmark() {
+  cyclictest_state=1
+  for i in {1..480}; do # 2hours
+    update
+    if [ "${benchmark_state}" == "Error" ]; then
+      log "Cerberus status is False, Cluster is unhealthy"
+      exit 1
+    fi
+    oc describe -n benchmark-operator benchmarks/cyclictest | grep State | grep Complete
+    if [ $? -eq 0 ]; then
+      log "cyclictest workload done!"
+      cyclictest_state=$?
+      break
+    fi
+    update
+    log "Current status of the ${WORKLOAD} benchmark is ${uline}${benchmark_state}${normal}"
+    check_logs_for_errors
+    sleep 30
+  done
+
+  if [ "$cyclictest_state" == "1" ] ; then
+    log "Workload failed"
+    exit 1
+  fi
+}
+
+assign_uuid() {
+  update
+  compare_testpmd_uuid=${benchmark_uuid}
+  if [[ ${COMPARE} == "true" ]] ; then
+    echo ${baseline_cyclictest_uuid},${compare_cyclictest_uuid} >> uuid.txt
+  else
+    echo ${compare_cyclictest_uuid} >> uuid.txt
+  fi
+}
+
+run_benchmark_comparison() {
+  log "Beginning benchmark comparison"
+  ../../utils/touchstone-compare/run_compare.sh cyclictest ${baseline_cyclictest_uuid} ${compare_cyclictest_uuid} 
+  log "Finished benchmark comparison"
+  }
+
+generate_csv() {
+  log "Generating CSV"
+  # tbd
+}
+
+delete_benchmark() {
+  oc delete benchmarks.ripsaw.cloudbulldozer.io/cyclictest -n benchmark-operator
+}
+
+update() {
+  benchmark_state=$(oc get benchmarks.ripsaw.cloudbulldozer.io/cyclictest -n benchmark-operator -o jsonpath='{.status.state}')
+  benchmark_uuid=$(oc get benchmarks.ripsaw.cloudbulldozer.io/cyclictest -n benchmark-operator -o jsonpath='{.status.uuid}')
+  benchmark_current_pair=$(oc get benchmarks.ripsaw.cloudbulldozer.io/cyclictest -n benchmark-operator -o jsonpath='{.spec.workload.args.pair}')
+}
+
+print_uuid() {
+  log "Logging uuid.txt"
+  cat uuid.txt
+}
+
+export TERM=screen-256color
+bold=$(tput bold)
+uline=$(tput smul)
+normal=$(tput sgr0)
+python3 -m pip install -r requirements.txt | grep -v 'already satisfied'
```

Comment:
We should probably deploy a venv first and then install/run from there to avoid any potential issues